crates/env/src/api.rs

```diff
@@ -526,6 +526,28 @@ where
     })
 }
 
+/// Returns the buffer back to the caller of the executed contract.
+///
+/// # Note
+///
+/// This function  stops the execution of the contract immediately.
+pub fn return_value_scoped(return_flags: ReturnFlags, return_value: &[u8]) -> ! {
```

Comment:
very confusing name imo. The scoped buffer concept is an internal low-level concept that should not be exposed to the user. I'd rename this to `return_bytes` since that's what it is.